src/network.rs

```diff
@@ -594,9 +594,12 @@ pub fn add(wifi: &WiFi) -> Result<(), NetworkError> {
             //  config file could also be copied from peach/config fs location
             Err(e) => panic!("Failed to write to file: {}", e),
         };
+        Ok(())
+    }
+    else {
+        let err_msg = String::from_utf8_lossy(&output.stdout);
+        Err(NetworkError::GenWpaPassphraseWarning{ ssid: wifi.ssid.to_string(), err_msg: err_msg.to_string()})
```

Comment:
You might be able to avoid an allocation here by passing `&wifi.ssid` as the value of `ssid`. And pass `err_msg` as it is (without additional `to_string()`):

```rust
// pass a reference to `wifi.ssid` (I believe `snafu` will do the string conversion in the formatter)
// for `err_msg`, the value is named the same as the field so you should have to do `err_msg: err_msg`
Err(NetworkError::GenWpaPassphraseWarning{ ssid: &wifi.ssid, err_msg})
```

P.s. Are you using the `clippy` tool? `cargo clippy` or `cargo +nightly clippy` (this often teaches me more efficient ways to write the same code)

Reply:
I tried the variation you suggested, and a few others, but all are giving me different compiler errors.. cant say I totally understand yet.. but the .to_string version is the only version I've found thats working